internal/ethapi/api.go

```diff
@@ -806,6 +820,13 @@ func DoEstimateGas(ctx context.Context, b Backend, args CallArgs, blockNr rpc.Bl
 // EstimateGas returns an estimate of the amount of gas needed to execute the
 // given transaction against the current pending block.
 func (s *PublicBlockChainAPI) EstimateGas(ctx context.Context, args CallArgs) (hexutil.Uint64, error) {
+	if gasCap := s.b.RPCGasCap(); gasCap != nil {
+		if new(big.Int).SetUint64(uint64(*args.Gas)).Cmp(gasCap) > 0 {
```

Comment:
args.Gas can be `nil`. This node will crash in that case. We handle gas illness inside `DoEstimateGas`. Perhaps we should pass the gascap as a parameter to `DoEstimateGas` and handle everything there.